### DISCUSSION TITLE
Headless option in the CLI

### DISCUSSION THREAD

**Discussion by cullzie** [aapatre/Automatic-Udemy-Course-Enroller-GET-PAID-UDEMY-COURSES-for-FREE#167] [Ideas]
Not everyone wants to have the browser window open when running the script. It would also allow people who may not have a system to run using github actions or by using some free tiers on server providers.

We could add headless option to CLI to enable this: `--headless`. 
Adding that functionality to the browsers would be required if we want to have a full CI of all the different browsers as well.

What do you think?

**Comment by fakeid30** [comment]
How would we go about achieving this ? Is there a firefox or chrome headless package? 

Yes the idea is good but I'm not sure how practical that is. How hard do you think it'll be ?

**Reply by cullzie** [reply]
Its just a case of passing the headless args to the browser on creation. Similar to what is currently done for chrome. It might take a while to test all browsers (mainly edge, internet explorer :vomiting_face:) but I've done it with firefox, chrome and chromium before.

**Comment by fakeid30** [comment]
then sure , its good. Btw the CI is failing with debug , idk why .

**Reply by cullzie** [reply]
Its not the debug but the login check. I don't think the account has been logging in for the past while looking at older logs.

**Comment by aapatre** [comment]
I once found a headless browser but it wasn't as robust as selenium... Also, didn't have the features to automate clicking

**Reply by cullzie** [reply]
Was it phatomJs by any chance? All modern browsers should have headless option built in. Just with headless it is harder to debug and can have some 'quirks'

**Comment by cullzie** [comment]
Locking this as we are moving away from browser usage for enrollment.
